lib/shoppertrak_api_client.py

```diff
@@ -0,0 +1,161 @@
+        except RequestException as e:
+            self.logger.error(f"Failed to retrieve response from {full_url}: {e}")
+            raise ShopperTrakApiClientError(
+                f"Failed to retrieve response from {full_url}: {e}"
+            ) from None
+
+        response_root = self._check_response(response.text)
+        if response_root is not None:
+            return response_root
+        elif query_count < self.max_retries:
+            self.logger.info("Waiting 5 minutes and trying again")
+            time.sleep(300)
+            return self.query(endpoint, date_str, query_count + 1)
+        else:
+            self.logger.error(
+                f"Reached max retries: sent {self.max_retries} queries with no response"
+            )
+            raise ShopperTrakApiClientError(
+                f"Reached max retries: sent {self.max_retries} queries with no response"
+            )
+
+    def parse_response(self, xml_root, input_date_str, is_recovery_mode=False):
+        rows = []
+        for site_xml in xml_root.findall("site"):
+            site_str = self._get_xml_str(site_xml, "siteID")
+            for date_xml in site_xml.findall("date"):
+                date_str = self._get_xml_str(date_xml, "dateValue")
+                if date_str != input_date_str:
+                    self.logger.error(
+                        f"Request date does not match response date.\nRequest date: "
+                        f"{input_date_str}\nResponse date: {date_str}"
+                    )
+                    raise ShopperTrakApiClientError(
+                        f"Request date does not match response date.\nRequest date: "
+                        f"{input_date_str}\nResponse date: {date_str}"
+                    )
+                for entrance_xml in date_xml.findall("entrance"):
+                    entrance_val = self._get_xml_str(entrance_xml, "name")
+                    if entrance_val:
+                        entrance_val = self._cast_str_to_int(entrance_val.lstrip("EP"))
+                    for traffic_xml in entrance_xml.findall("traffic"):
+                        enters = self._cast_str_to_int(
+                            self._get_xml_str(traffic_xml, "enters")
+                        )
+                        exits = self._cast_str_to_int(
+                            self._get_xml_str(traffic_xml, "exits")
+                        )
+
+                        start_time_str = self._get_xml_str(traffic_xml, "startTime")
+                        start_dt_str = datetime.strptime(
+                            date_str + " " + start_time_str, "%Y%m%d %H%M%S"
+                        ).isoformat()
+
+                        code = self._get_xml_str(traffic_xml, "code")
+                        is_healthy_orbit = code == "01"
+                        if code != "01" and code != "02":
+                            self.logger.warning(
+                                f"Unknown code: '{code}'. Setting is_real to False"
```

Comment:
What is `is_real`? I only see this reference to it

Reply:
Whoops this was from an old variable name -- updated!